### PR TITLE
add cors support for /api

### PR DESCRIPTION
**Pull request by abhishek97**
In reference to : #30

### CODE REVIEW

File: routers/api/index.js [hunk 1]
```diff
@@ -4,7 +4,14 @@
  * This is the /api/v1 path
  */
 const router = require('express').Router();
+const cors = require('cors');
 
+const options = {
+    origin : '*',
```
Comment (championswimmer) [inline routers/api/index.js]: origin should not be *
it should be same as the domain of the client.

Comment (championswimmer) [inline routers/api/index.js]: Check http://stackoverflow.com/questions/1653308/access-control-allow-origin-multiple-origin-domains

Comment (abhishek97) [inline routers/api/index.js]: Client as in who is making the api request? 
So if anyone who makes an api request gets a suitable CORS header then why bother with the header ? 
Is there an attack vector here?

Comment (championswimmer) [inline routers/api/index.js]: Check the domain from where request has come from
Find client by clientId (present in request), and verify that it _contains_ the domain in it's domains column
If yes, return with the same domain in the CORS field. 

If I get COR from a.com, with client id 1234. Then `Client.findById(1234).getValueOfColumn('domains').contains(req.referrer) ?`

Comment (abhishek97) [inline routers/api/index.js]: @championswimmer  We need cors headers only on `/api` endpoints. As `/oauth` will never be made request to ( from a browser ) . So we can either 
1. set cors to match the domain of client (which we get from auth-token in header of request)
2. Have `/api` as origin. Which is not bad as there is already authentication of auth-token done.

If we go the first way, we break some frontend part of oauth where we have api request from same domain and no `Authorisation` header.

Comment (championswimmer) [inline routers/api/index.js]: We need CORS headers on /api endpoints. 

When we get request. We check token, find out which client send request. If the requesting referrer is present in the array of domains of the client, we echo back the domain, else we do not.

Comment (abhishek97) [inline routers/api/index.js]: @championswimmer then the frontend which is cosuming the api like this : https://github.com/coding-blocks/oneauth/blob/master/views/client/add.hbs#L5 
shouldn't work, then i would need to enforce auth-token checking.

This logic will work : 

```
if ( req.origin == process.env.origin ) {
 next();
} else  {
  // authenticate token and get client
   passport.authenticate('bearer').call(this,req,res,addCorsHeaders ); 
}

```

But seems like a anti-pattern. Is this be okay?

**Pull request closed without merging**